hackathon_contributions_new_data_format/Isensee_JCB2018/visualization/main.py

```diff
@@ -0,0 +1,187 @@
+
+        # define index to reduce measurement_data to data linked to datasetId
+        ind_dataset = measurement_data['datasetId'] == datasetId
+
+        # gather simulationConditionIds belonging to datasetId
+        uni_conditionIds = np.unique(measurement_data[ind_dataset].simulationConditionId)
+        clmn_name_unique = 'simulationConditionId'
+
+        # Case seperation indepParameter custom, time or condition
+        if indepVar not in ['time', "condition"]:
+
+            # extract conditions (plot input) from condition file
+            ind_cond = experimental_condition['conditionId'].isin(uni_conditionIds)
+            conditions = experimental_condition[ind_cond][indepVar]
+
+            # # create empty dataframe for means and SDs
+            # ms = pd.DataFrame(columns = ['mean', 'sd'], index=conditionIds)
+            #
+            # # group measurement values for each conditionId
+            # for ID in conditionIds:
+            #     meas = measurement_data['simulationConditionId'] == ID
+            #
+            #     ms.at[ID,'mean'] = np.mean(measurement_data[meas].measurement)
+            #     ms.at[ID,'sd'] = np.std(measurement_data[meas].measurement)
+
+            # group measurement values for each conditionId
+            ms = getDataToBePlotted.getDataToBePlotted(visualization_specification, measurement_data, uni_conditionIds,
+                                                       i, clmn_name_unique)
+
+            # set xScale
+            if visualization_specification.xScale[i] == 'lin':
+                ax[axx, axy].set_xscale("linear")
+            elif visualization_specification.xScale[i] == 'log10':
+                ax[axx, axy].set_xscale("log")
+            elif visualization_specification.xScale[i] == 'order':        # equidistant
+                ax[axx, axy].set_xscale("linear")
+                # check if conditions are monotone decreasing or increasing
+                if np.all(np.diff(conditions) < 0):             # monotone decreasing
+                    xlabel = conditions[::-1]                   # reversing
+                    conditions = range(len(conditions))[::-1]   # reversing
+                    ax[axx, axy].set_xticks(range(len(conditions)), xlabel)
+                elif np.all(np.diff(conditions) > 0):
+                    print('monotone increasing')
+                    xlabel = conditions
+                    conditions = range(len(conditions))
+                    ax[axx, axy].set_xticks(range(len(conditions)), xlabel)
+                else:
+                    print('Error: x-conditions do not coincide, some are mon. increasing, some monotonically decreasing')
+
+
+            if visualization_specification.plotTypeData[i] == 'MeanAndSD':
+                ax[axx, axy].errorbar(conditions, ms['mean'], ms['sd'], linestyle='-', marker='.',
+                                        color = cmap[min(7, i - plotInd[i_plotId])],
```

Comment:
Avoid magic numbers. Where does 7 come from?

Reply:
comes from the color array, but ist resolved with the proper implementation of colormaps